deploy/crds/nodemaintenance_crd.yaml

```diff
@@ -17,23 +16,51 @@ spec:
     openAPIV3Schema:
       properties:
         apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
           type: string
         kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
           type: string
         metadata:
           type: object
         spec:
           properties:
             nodeName:
+              description: Node name to apply maintanance on/off
               type: string
             reason:
+              description: Reason for maintanance
               type: string
           required:
           - nodeName
           type: object
         status:
           properties:
-            phase:
+            completed:
+              description: Completed represents the state of maintenance. True if
+                completed , false otherwise
+              type: boolean
+            errorMessage:
+              description: ErrorMessage represents the latest reason for failed=true
+              type: string
+            failed:
+              description: Failed is true if the controller did not complete maintenance
+                during the latest reconciliation, false otherwise
+              type: boolean
+            pendingPods:
+              description: PendingPods are pods that failed to be evicted in the latest
+                reconciliation
+              items:
+                type: object
+              type: array
+            startTime:
+              description: StartTime contains the Date and time at which the maintenance
```

Comment:
Is this very much different from the CreationTimestamp we already get in the metadata section?

Reply:
I was contemplating about this part, what is actually the start time.
Is it when the CR created (which makes sense) or is it when we actually started reconciling on it.
I now tend to remove this field and rely on the creationTimestamp.
@jtomasek  any thoughts on that from your perspective as a UI consumer ?

Reply:
I'd consider the maintenance as initiated once the CR is created, so yes `creationTimestamp` should be good enough.